ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/CancelFunctionButton/CancelFunctionModal.tsx

```diff
@@ -0,0 +1,181 @@
+      queuedAtMax: timeRange.end,
+    };
+  }
+
+  const countRes = useRunCount(runCountInput);
+  const featureRes = usePlanFeatures();
+
+  let upgradeCutoff: Date | undefined;
+  if (featureRes.data) {
+    upgradeCutoff = subtractDuration(new Date(), { days: featureRes.data.history });
+  }
+
+  return (
+    <Modal className="w-[800px]" isOpen={isOpen} onClose={onClose}>
+      <Modal.Header>Bulk cancellation</Modal.Header>
+
+      <Modal.Body>
+        <div className="flex flex-col gap-4">
+          <div>
+            <Input
+              label="Name"
+              name="cancellation-name"
+              onChange={(e) => {
+                setName(e.target.value);
+              }}
+              optional
+              placeholder="My cancellation"
+              value={name}
+            />
+          </div>
+
+          <div>
+            <label className="text-basis text-sm font-medium" htmlFor="time-range">
+              Date range
+            </label>
+            <p className="text-subtle text-sm">
+              Choose the time range when the function runs were queued
+            </p>
+
+            <RangePicker
+              className="w-full"
+              disabled={featureRes.isLoading}
+              onChange={(range) =>
+                setTimeRange(
+                  range.type === 'relative'
+                    ? { start: subtractDuration(new Date(), range.duration), end: new Date() }
+                    : { start: range.start, end: range.end }
+                )
+              }
+              upgradeCutoff={upgradeCutoff}
+            />
+          </div>
+
+          <Alert severity="info">
+            This action will affect only queued and running function runs. All affected function
+            runs will immediately cancel, but their status may not update immediately.
+          </Alert>
+
+          {countRes.error && (
+            <Alert severity="error">Failed to query run count: {countRes.error.message}</Alert>
+          )}
+
+          {featureRes.error && (
+            <Alert severity="error">
+              Failed to query plan features: {featureRes.error.message}
+            </Alert>
+          )}
+
+          {creationError && (
+            <Alert severity="error">Failed to create cancellation: {creationError.message}</Alert>
+          )}
+        </div>
+      </Modal.Body>
+
+      <Modal.Footer className="flex gap-2">
+        <div className="grow">
+          <FooterMessage count={countRes.data} />
+        </div>
+
+        <NewButton
+          appearance="outlined"
+          disabled={isCreating}
+          kind="secondary"
+          label="Close"
+          onClick={onClose}
+        />
+        <NewButton
+          disabled={isCreating || !timeRange || countRes.data === 0}
+          kind="danger"
+          label="Submit"
+          onClick={onSubmit}
+        />
+      </Modal.Footer>
+    </Modal>
+  );
+}
+
+function FooterMessage({ count }: { count: number | undefined }) {
```

Comment:
I think a typescript optional is a bit simpler. Also seems like a nice case (dad joke!) for a switch! Like:

```
function FooterMessage({ count }: { count?: number }) {
  switch (count) {
    case undefined:
      return null;
    case 0:
      return <span>No runs to cancel</span>;
    case 1:
      return <span>Approximately 1 run will cancel</span>;
    default:
      <span>Approximately {count} runs will cancel</span>;
  }
}
```

Reply:
`count?: number` is different than `count: number | undefined`, since the former makes the `count` _optional_ rather than _possible undefined_. If you make it optional, then a consumer may forget to pass it.

As for the switch, that's a personal preference thing and I dislike switches